gammapy/maps/wcs/geom.py

```diff
@@ -114,8 +114,8 @@ def __init__(self, wcs, npix=None, cdelt=None, crpix=None, axes=None):
         self._crpix = crpix
 
         # define cached methods
-        self.get_coord = lru_cache()(self.get_coord)
-        self.get_pix = lru_cache()(self.get_pix)
+        self.get_coord = lru_cache()(self._get_coord)
+        self.get_pix = lru_cache()(self._get_pix)
```

Comment:
I don't see improvement we the proposed solution but I think we can fix it by forcing garbage collection here: 
```suggestion
        gc.collect()
        self.get_coord = lru_cache()(self.get_coord)
        self.get_pix = lru_cache()(self.get_pix)
```

For my test I run
```
from gammapy.datasets import MapDataset
from regions import CircleSkyRegion
import astropy.units as u

dataset = MapDataset.read( "test1.fits")
empty = MapDataset.from_geoms(**dataset.geoms)
reg = CircleSkyRegion(center=empty._geom.center_skydir, radius=1.0*u.deg)

N=10
for i in range(N):
    ds = MapDataset.read("test1.fits")  
    ds.mask_fit = ds._geom.region_mask([reg], inside=False) 

```
and I get for this PR
![pr](https://github.com/user-attachments/assets/b7d63c06-592a-4b88-a398-b8124c0d59f3)

while adding gc.collect just before defining the cached method
![geom_collect](https://github.com/user-attachments/assets/29fc6beb-bd8c-4ba2-a775-6cd12a81745a)